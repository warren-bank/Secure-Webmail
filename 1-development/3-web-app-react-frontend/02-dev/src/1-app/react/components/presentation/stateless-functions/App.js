const React       = require('react')
const PropTypes   = require('prop-types')

const purify      = require('react/components/higher-order/purify')
const displayName = 'App'

const Header      = require(`./${displayName}/Header`)
const Sidebar     = require(`./${displayName}/Sidebar`)

const component = ({state, component}) => {
  return (
    <div className={`top-component ${displayName.toLowerCase()}`}>
      <Header user={state.user} />
      <Sidebar folders={state.folders} active_folder={state.app.ui.folder_name} />
      {component}
    </div>
  )
}

component.propTypes = {
  state:      PropTypes.object.isRequired,
  component:  PropTypes.instanceOf(React.Component).isRequired
}

component.displayName = displayName

module.exports = purify(component)
