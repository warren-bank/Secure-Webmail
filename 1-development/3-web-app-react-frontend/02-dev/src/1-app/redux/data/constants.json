{
    "actions": {
        "STORE_INITIALIZED":                "STORE_INITIALIZED",
        "LOG_DEBUG_MESSAGE":                "LOG_DEBUG_MESSAGE",
        "PROPOGATE_SERVER_API_ERROR":       "PROPOGATE_SERVER_API_ERROR",

        "GET_FOLDERS":                      "GET_FOLDERS",
        "SAVE_FOLDERS":                     "SAVE_FOLDERS",

        "GET_THREADS_IN_FOLDER":            "GET_THREADS_IN_FOLDER",
        "SAVE_THREADS_TO_FOLDER":           "SAVE_THREADS_TO_FOLDER",
        "SAVE_THREADS":                     "SAVE_THREADS",

        "GET_THREAD":                       "GET_THREAD",
        "SAVE_THREAD":                      "SAVE_THREAD",

        "UPDATE_THREAD":                    "UPDATE_THREAD",
        "SAVE_THREAD_UPDATE":               "SAVE_THREAD_UPDATE",

        "UPDATE_MESSAGE":                   "UPDATE_MESSAGE",
        "SAVE_MESSAGE_UPDATE":              "SAVE_MESSAGE_UPDATE",

        "SET_RSA_PUBLIC_KEY":               "SET_RSA_PUBLIC_KEY",
        "GET_RSA_PUBLIC_KEYS":              "GET_RSA_PUBLIC_KEYS",
        "SAVE_RSA_PUBLIC_KEYS":             "SAVE_RSA_PUBLIC_KEYS",

        "UPDATE_SETTINGS":                  "UPDATE_SETTINGS",
        "SAVE_APP": {
            "UI": {
                "FOLDER_NAME":              "SAVE_APP_UI_FOLDER_NAME",
                "THREAD_ID":                "SAVE_APP_UI_THREAD_ID",
                "START_THREADS_INDEX":      "SAVE_APP_UI_START_THREADS_INDEX"
            },
            "SETTING": {
                "DISPLAY_HTML_FORMAT":      "SAVE_APP_SETTING_DISPLAY_HTML_FORMAT",
                "COMPOSE_HTML_FORMAT":      "SAVE_APP_SETTING_COMPOSE_HTML_FORMAT",
                "MAX_THREADS_PER_PAGE":     "SAVE_APP_SETTING_MAX_THREADS_PER_PAGE",
                "PUBLIC_KEY":               "SAVE_APP_SETTING_PUBLIC_KEY",
                "PRIVATE_KEY":              "SAVE_APP_SETTING_PRIVATE_KEY",
                "PRIVATE_KEY_STORAGE":      "SAVE_APP_SETTING_PRIVATE_KEY_STORAGE",
                "IS_GENERATING_KEYPAIR":    "SAVE_APP_SETTING_IS_GENERATING_KEYPAIR"
            },
            "DRAFT_MESSAGE": {
                "STORE":                    "SAVE_APP_DRAFT_MESSAGE_STORE",
                "CLEAR":                    "SAVE_APP_DRAFT_MESSAGE_CLEAR",
                "SET_STATUS":               "SAVE_APP_DRAFT_MESSAGE_SET_STATUS"
            }
        },
        "CRYPTO": {
            "RSA": {
                "GENERATE_KEYPAIR":         "CRYPTO_RSA_GENERATE_KEYPAIR"
            },
            "AES": {
            }
        },

        "SEND_EMAIL": {
            "REPLY":                        "SEND_EMAIL_REPLY",
            "NEW_MESSAGE":                  "SEND_EMAIL_NEW_MESSAGE"
        },

        "RESPOND_TO_USER_EVENT": {
            "REDIRECT_URL":                 "USER_EVENT_REDIRECT_URL",
            "OPEN_FOLDER":                  "USER_EVENT_OPEN_FOLDER",
            "OPEN_THREAD":                  "USER_EVENT_OPEN_THREAD",
            "OPEN_COMPOSE_MESSAGE":         "USER_EVENT_OPEN_COMPOSE_MESSAGE",
            "DOWNLOAD_PRIVATE_KEY":         "DOWNLOAD_PRIVATE_KEY"
        }
    },
    "storage": {
        "PRIVATE_KEY":                      "PRIVATE_KEY"
    },
    "encryption": {
        "RESERVED_ATTACHMENT_NAME": {
            "CIPHERS":                      "CIPHERS",
            "BODY":                         "BODY"
        },
        "CLEARTEXT_CONTENT": {
            "BODY":                         "Encrypted by: Secure Webmail\n\nhttps://secure-webmail.github.io/"
        }
    },
    "default_settings": {
        "display_html_format":              true,
        "compose_html_format":              true,
        "max_threads_per_page":             25,
        "public_key":                       "",
        "private_key":                      "",
        "private_key_storage":              0,
        "is_generating_keypair":            false
    },
    "timer_periods": {
        "reply": {
            "thread_update":                5000
        }
    },
    "export_filenames": {
        "PRIVATE_KEY":                      "Secure-Webmail.private-key.txt"
    }
}
