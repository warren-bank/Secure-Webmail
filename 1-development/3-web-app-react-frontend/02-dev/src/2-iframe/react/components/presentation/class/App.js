const React       = require('react')
const PropTypes   = require('prop-types')

const {GoogleLogin, GoogleLogout} = require('react-google-login')

const addEventListener  = require('react/lib/loginParentIframe')
const constants         = window.constants

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {account: null}

    this.default_login_props = {
      clientId:         constants.clientId,
      scope:            "profile email",
      onSuccess:        this.onSuccess.bind(this),
      onFailure:        this.onFailure.bind(this)
    }

    this.buttons = {
      login: (
        <GoogleLogin  className="google login"  {...this.default_login_props} prompt="consent" >
          <span className="icon"></span>
          <span class="buttonText">Login</span>
        </GoogleLogin>
      )
    }

    this.tid = (() => {
      if (window.location.search) {
        const search_pattern = /[\?&]tid=([^&]+)(?:&|$)/
        const matches        = search_pattern.exec(window.location.search)

        if (matches && Array.isArray(matches)) {
          window.history.replaceState(null, null, window.location.pathname)
          return matches[1]
        }
      }
      return null
    })()

    addEventListener( this.setAccount.bind(this) )
  }

  setAccount(account) {
    this.setState({account})
  }

  onSuccess(response) {
    console.log('onSuccess:', response)

    this.setAccount(response.profileObj.email)
  }

  onFailure(response) {
    console.error('onFailure:', response)
  }

  render() {
    let url    = constants.urls.iframe_source

    if (url.indexOf('data:text/html;base64,') !== 0) {
      let params = []
      if (this.tid)
        params.push(`tid=${window.encodeURIComponent(this.tid)}`)
      if (this.state.account)
        params.push(`email=${window.encodeURIComponent(this.state.account)}`)
      if (params.length) {
        params = params.join('&')
        url += '?' + params
      }
    }

    return (
      <div className="app">
      {
        !this.state.account &&
        <div className="header">
          {this.buttons.login}
        </div>
      }
        <div className="iframe_container">
          <iframe
            id={this.props.iframe_id}
            src={url}
            frameborder="0"
            width="100%"
            marginheight="0"
            marginwidth="0"
            scrolling="no"
          ></iframe>
        </div>
      </div>
    )
  }
}

App.propTypes = {
  iframe_id: PropTypes.string.isRequired
}

module.exports = App
