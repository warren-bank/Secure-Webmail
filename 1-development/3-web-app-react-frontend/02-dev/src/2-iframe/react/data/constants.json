{
    "clientId": "133752900170-96kd1367qdladljtl6isd7i2mo2pvi0c.apps.googleusercontent.com",
    "urls": {
        "iframe_parent": "https://secure-webmail.github.io/",
        "iframe_source": "https://script.google.com/macros/s/AKfycbxjGr_DXhEsXfQxDtKII3LkCEGC5zgsQcdx4DJNWJitkyx8FLZV/exec"
    }
}
