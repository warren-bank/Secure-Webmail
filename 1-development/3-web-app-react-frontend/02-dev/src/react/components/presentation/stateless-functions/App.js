import React      from 'react'
import PropTypes  from 'prop-types'

import purify     from 'react/components/higher-order/purify'
import AddTodo    from './AddTodo'
import TodoList   from './TodoList'
import Footer     from './Footer'

const getVisibleTodos = (todos, filter, C) => {
  switch (filter) {
    case C.SHOW_COMPLETED:
      return todos.filter(t => t.completed)
    case C.SHOW_ACTIVE:
      return todos.filter(t => !t.completed)
    case C.SHOW_ALL:
    default:
      return todos
  }
}

const App = ({state}, {constants}) => {
  let C     = constants.filter.values
  let todos = getVisibleTodos(state.todos, state.visibilityFilter, C)

  return (
    <div>
      <AddTodo />
      <TodoList todos={todos} />
      <Footer visibilityFilter={state.visibilityFilter} />
    </div>
  )
}

App.propTypes = {
  state: PropTypes.object.isRequired
}

App.displayName = 'App'

export default purify(App)
