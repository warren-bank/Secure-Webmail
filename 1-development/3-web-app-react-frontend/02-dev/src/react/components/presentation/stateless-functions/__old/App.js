const React      = require('react')
const PropTypes  = require('prop-types')

const purify     = require('react/components/higher-order/purify')
const AddTodo    = require('./AddTodo')
const TodoList   = require('./TodoList')
const Footer     = require('./Footer')

const App = ({state}, {constants}) => {
  return (
    <div>
      <AddTodo />
      <TodoList todos={state.todos} />
      <Footer visibilityFilter={state.visibilityFilter} />
    </div>
  )
}

App.propTypes = {
  state: PropTypes.object.isRequired
}

App.displayName = 'App'

module.exports = purify(App)
