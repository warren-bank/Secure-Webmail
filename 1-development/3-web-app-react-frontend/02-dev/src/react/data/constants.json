{
    "filter": {
        "values":  {
            "SHOW_ALL": "SHOW_ALL",
            "SHOW_ACTIVE": "SHOW_ACTIVE",
            "SHOW_COMPLETED": "SHOW_COMPLETED"
        },
        "default": "SHOW_ALL"
    }
}
