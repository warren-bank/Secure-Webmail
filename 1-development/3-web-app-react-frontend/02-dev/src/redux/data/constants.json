{
    "actions": {
        "ADD_TODO": "ADD_TODO",
        "TOGGLE_TODO": "TOGGLE_TODO",
        "SET_VISIBILITY_FILTER": "SET_VISIBILITY_FILTER"
    },
    "middleware": {
        "LOCAL_STORAGE": "react-redux-bindings-todo-list-app"
    }
}
