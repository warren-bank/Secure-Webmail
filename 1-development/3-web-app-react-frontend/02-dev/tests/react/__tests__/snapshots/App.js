import App       from 'App'
import constants from 'react/data/constants'

const {initialState} = redux
const {expectJSX}    = enzyme

describe('[snapshot] React stateless presentation component: App', function() {

  it('should render static JSX for the initial Redux state', function() {
    let component = <App state={initialState} />
    let context   = {constants}

    expectJSX(component, {context}).toMatchSnapshot()
  })

})
