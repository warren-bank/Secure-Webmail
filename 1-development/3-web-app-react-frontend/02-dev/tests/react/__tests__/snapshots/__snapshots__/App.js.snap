// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshot] React stateless presentation component: App should render static JSX for the initial Redux state 1`] = `
<div>
  <AddTodo />
  <TodoList
    todos={
      Array [
        Object {
          "completed": false,
          "id": "a3b96f13-0562-4a6b-8163-e851e823cfda",
          "text": "milk",
        },
        Object {
          "completed": true,
          "id": "68f4e07e-e755-41cb-ae89-5d77096a8b06",
          "text": "eggs",
        },
        Object {
          "completed": false,
          "id": "177da1a5-71a8-4746-8f42-3b4f79518688",
          "text": "cheese",
        },
      ]
    }
  />
  <Footer
    visibilityFilter=""
  />
</div>
`;
